Applications/Spire/Source/KeyBindings/CancelKeyBindingsTableModel.cpp

```diff
@@ -0,0 +1,171 @@
+#include "Spire/KeyBindings/CancelKeyBindingsTableModel.hpp"
+
+using namespace Spire;
+
+namespace {
+  const auto ROW_COUNT = 13;
+  const auto COLUMN_COUNT = 2;
+
+  auto get_action(int index) {
+    static auto actions = std::vector<KeyBindings::CancelAction>({
+      KeyBindings::CancelAction::ALL,
+      KeyBindings::CancelAction::ALL_ASKS,
+      KeyBindings::CancelAction::ALL_BIDS,
+      KeyBindings::CancelAction::CLOSEST_ASK,
+      KeyBindings::CancelAction::CLOSEST_BID,
+      KeyBindings::CancelAction::FURTHEST_BID,
+      KeyBindings::CancelAction::FURTHEST_ASK,
+      KeyBindings::CancelAction::MOST_RECENT,
+      KeyBindings::CancelAction::MOST_RECENT_ASK,
+      KeyBindings::CancelAction::MOST_RECENT_BID,
+      KeyBindings::CancelAction::OLDEST,
+      KeyBindings::CancelAction::OLDEST_ASK,
+      KeyBindings::CancelAction::OLDEST_BID
+    });
+    return actions[index];
+  }
+
+  auto get_index(KeyBindings::CancelAction action) {
```

Comment:
Ask Kamal on whether it is acceptable to use enum's underlying types directly for indexing. It may be considered unstable but, when allowed, it saves a lot of trouble.

Reply:
Yes it is, at any rate using a switch is better.

Reply:
Assuming it's acceptable does the ordering you've declared the CancelAction enum make a difference to you?

Reply:
The order should not make any difference to anyone. It is a plain enum, as in a set of distinct values identified by a sequence of integers.
If `switch` is better, then use `switch`. You could avoid these functions by casting the enum type directly to int and vice versa, but it is not very stable as even a minor change to the API may break the implementation. For example, one could assign a non-zero value to the first element of the enum for whatever reason.